### PR TITLE
add MSR ANY hooks, flt subsystem and cr4 checks

### PR DESCRIPTION
**Pull request by archercreat**
Hello! This pr adds ability to hook specific msr registers, cr4 smap smep and fltmgr subsystem checks.

### CODE REVIEW

File: src/plugins/rootkitmon/rootkitmon.cpp [hunk 1]
```diff
@@ -563,7 +554,7 @@ static void driver_visitor(drakvuf_t drakvuf, addr_t driver, void* ctx)
 
     // Map 1 4KB page with PE header
     void* module = nullptr;
-    if (VMI_SUCCESS != vmi_mmap_guest(vmi, &a_ctx, 1, &module) || !module )
+    if (VMI_SUCCESS != vmi_mmap_guest(vmi, &a_ctx, 1, &module))
```
Comment (tklengyel) [inline src/plugins/rootkitmon/rootkitmon.cpp:557]: Don't revert this change here, it was causing CI failures.

**Comment by tklengyel** [comment]
@drakvuf-jenkins Test this please

**Comment by tklengyel** [comment]
Could you rebase this on the latest main? For some reasom the CI didn't trigger

**Comment by tklengyel** [comment]
@drakvuf-jenkins Test this please

**Comment by archercreat** [comment]
Sorry, looks like unwanted commits got included as well. I will reopen and split requests now.

**Pull request closed without merging**